dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -310,7 +318,10 @@ void CDefaultContextMenu::AddStaticEntriesForKey(HKEY hKey)
         if (RegEnumKeyExW(hShellKey, dwIndex++, wszName, &cchName, NULL, NULL, NULL, NULL) != ERROR_SUCCESS)
             break;
 
+        RegCloseKey(hShellKey);
         AddStaticEntry(hKey, wszName);
+        if(RegOpenKeyExW(hKey, L"shell", 0, KEY_READ, &hShellKey) != ERROR_SUCCESS)
```

Comment:
What's the purpose of this? This seems to be opening the same key, when AddStaticEntry doesn't make any modifications to it.